### PR TITLE
Initial commit of calculator

### PR DESCRIPTION
**Pull request by kschumy**
# Calculator
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Question 	|  Answer 	|
|---	|---	|
|  Describe how you stored user input in your program?	|   I  took them in as a string, traversed each char of the string, and evaluated the chars	|
|  How did you determine what operation to perform?	|  I based this on the operation character 	|
|  Do you feel like you used consistent indentation throughout your code? 	|   I think so	|
|  If you had more time, what would you have added to or changed about the program? 	|   It doesn't work in some cases and I'd like to fixed these. Example 3timestimes2 = busted :(	|

### CODE REVIEW

**Comment by kschumy** [comment]
Also my comments are crappy. I would have liked to improve this I have I had more time. I need to be better about doing these as I go. I always end up changing things so much, but still, changing a ton of stuff isn't a good excuse to put off doing these.

File: simple_calculator_non_biexptree.rb [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+# Ada[9], Ampers
+# Week 1: Calculator
+# February 8, 2018
+#
+# This program prompts the user for an equation and prints the solution.
+
+
+# This program takes in operation and number characters and returns all the
+# stored characters
+class EquationBuilder
```
Comment (CheezItMan) [inline simple_calculator_non_biexptree.rb:11]: interesting a class

File: simple_calculator_non_biexptree.rb [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+# otherwise, prints message about provided equation being broken and then
+# quits program.
+#   Example -
+#     no error:    equation = [num_one, 'operator', num_two]
+#     error:       equation = [num_one, 'operator', 'other operator']
+#
+# Evaluates provided equation and returns results
+def calculate(equation)
+  check_if_invalid_equation(equation)
+  num_one, operator, num_two = equation
```
Comment (CheezItMan) [inline simple_calculator_non_biexptree.rb:226]: nice way to pull things off the array

File: simple_calculator_non_biexptree.rb [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+#
+# This program prompts the user for an equation and prints the solution.
+
+
+# This program takes in operation and number characters and returns all the
+# stored characters
+class EquationBuilder
+  # Creates a new EquationBuilder
+  def initialize
+      @new_number = ""  # stores incomplete number characters
```
Comment (CheezItMan) [inline simple_calculator_non_biexptree.rb:14]: a little too much indentation, minor thing

**Comment by CheezItMan** [comment]
# Calculator
## What We're Looking For

Feature  | Feedback
---       | ---
Takes in two numbers and an operator and performs the mathematical operation. | Goes further to take in an equation plus parenthese
Readable code with consistent indentation. | Some minor indentation issues, overall pretty good.
Summary | Outstanding work, great job with the extras handling order of operations, parentheses even!

File: simple_calculator_non_biexptree.rb [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+  def get_multiply_or_exponent
+    @equation_array.last == "*" ? (return "*#{@equation_array.pop}") :
+      (return "*")
+  end
+
+  # Pre: throws ArgumentError if @new_operator is an invalid operation.
+  #
+  # Returns the converted complete @new_operator
+  def get_converted_operator
+    case @new_operator
```
Comment (CheezItMan) [inline simple_calculator_non_biexptree.rb:110]: clever

File: simple_calculator_non_biexptree.rb [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+end
+
+# Solves provided equation_array and return solution
+def do_math(equation_array, operators)
+  return equation_array if !equation_array.any?(operators) ||
+    equation_array.length < 3
+  equation_array.each_index do |index|
+    if has_operator_as_next_char(equation_array[index + 1], operators)
+      equation_array.insert(index, calculate(equation_array.slice!(index, 3)))
+      return do_math(equation_array, operators)
```
Comment (CheezItMan) [inline simple_calculator_non_biexptree.rb:290]: Very advanced use of a technique called recursion.  Nice!

**Pull request closed without merging**